### PR TITLE
feat(tests): enable parallelization

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#9093 
#8814 

### Description
This PR introduces the parallelization feature for Cypress testing. 
The work for this PR consisted in creating a IBM Cloud CodeEngine instance out of an image provided by the sorry-cypress service, especifically the `sorry-cypress-director` service. The Director instance is the one who controls what tests each machine should run, coming from the same spec file run.

In order to achieve parallelization among the newly created machine matrix, we are now using the `cy2` dependency, which intercepts the parallelization call made to Cypress Dashboard and redirects all the data to our CE instance. 

However, in order to speed things up and ensure each machine doesn't need to install and build every time, we separate the workflow into separate pieces:

1. Install job
   - Install dependencies 
   - Cache dependencies
   - Build storybook
   - Upload `storybook` artifacts
2. Run tests
   - Retrieve cached dependencies
   - Retrieve `storybook` artifacts
   - Run tests
   - If one test fails, create a `failure.txt` file and upload as artifact
3. Check if all tests passed
   - Wait until all tests finish
   - Retrieve `failure.txt` artifact
   - If said file exists, fail this job

The caching and artifact uploading ensures that each of the test jobs don't need to install/rebuild everything all over again. With this setup, that should only happen once, speeding up the rest of the jobs.

### Changelog

**New**

- introduced parallelization 

**Changed**

- `e2e-tests.yml` file

### CODE REVIEW

File: packages/react/tests/e2e-storybook/cypress/fixtures/components.json [hunk 1]
```diff
@@ -0,0 +1,242 @@
+[
```
Comment (kennylam) [inline packages/react/tests/e2e-storybook/cypress/fixtures/components.json:1]: Pretty sure this file should be added to `.gitignore`?

File: packages/web-components/tests/e2e-storybook/cypress/fixtures/components.json [hunk 2]
```diff
@@ -0,0 +1,254 @@
+[
```
Comment (kennylam) [inline packages/web-components/tests/e2e-storybook/cypress/fixtures/components.json:1]: Pretty sure this file should be added to `.gitignore`?

**Review by kennylam — COMMENTED** [review]
Nice one @IgnacioBecerra! Do you think this would be ready to go as-is, or is more optimization/testing needed on CodeEngine before we merge?

**Comment by IgnacioBecerra** [comment]
@kennylam Due to the way parallelization occurs, those files need to be already existing since the Storybook doc test needs it right away. If it doesn't exist yet the test will fail right away. Figured that since we already have the other fixtures there and we haven't had a new component in a while it should be fine.

As for merging this, I want to keep running the tests for just a bit longer in this PR to ensure things are still working/passing before having them run in all of our PRs.

**Comment by m4olivei** [comment]
Hey there :wave: I came across this PR working in the `feat/masthead-v2` family of feature branches where I was cleaning up tests there to sync with changes. I started looking at the timing, specifically the `e2e-tests` job, and ended up finding this PR.

I'm wondering where this work is at? Also wondering, if we split the caching and parallizing pieces, could we get caching as an easy win and split paralliation to a follow up? Just a thought, not knowing much about the work involved here, and being very new to Github actions (coming from CircleCI).

**Comment by kennylam** [comment]
@m4olivei We've had to put this work on hold for a bit, but it's definitely something we want to get done in the future. Your suggestions are great and hopefully we can implement something along those lines once we streamline the process and see the net benefits from our first real world tests.

**Comment by IgnacioBecerra** [comment]
Closing in favor of #10314

**Pull request closed without merging**